### ISSUE TITLE
lynx not loading with wan 2.2 Fun VACE

### ISSUE THREAD

**Issue by Rudra-ai-coder** [kijai/ComfyUI-WanVideoWrapper#1413]
although added in the extra model the error shows Lynx ip embeds error needs an adapter, this is for the wan 2.2 high block

**Comment by irrikrlla** [comment]
lynx is based on wan2.1, not wan2.2.

**Comment by hbarham** [comment]
could we get lynx with wan2.2 sometime soon?

**Comment by Rudra-ai-coder** [comment]
> lynx is based on wan2.1, not wan2.2.

i know that, so does many other LORAs the architecture is same and Wan 2.2 is an extended training on Wan 2.1, so lynx should be usable architecturally, in the initial update of lynx it did work but after the recent updates the above error

**Comment by kijai** [comment]
Can't reproduce with main branch at least, just adding VACE to the example workflow works, and 2.2 doesn't load it any different.

**Comment by Rudra-ai-coder** [comment]
ok we need to connect vace also to the prev node of the extra select node of the ip and ref extra model select nodes

**Comment by Rudra-ai-coder** [comment]
if we connect vace on the vace node the ip and ref models don't load, probably because both are extra model loaders

**Comment by Rudra-ai-coder** [comment]
now its working consistency of the faces has increased tremendously

**Comment by DavidJBarnes** [comment]
Adding a data point for anyone who lands here from search, since this thread is the main hit for "Lynx + Wan2.2".

Confirming @kijai's "2.2 doesn't load it any different" from a different angle: I got the Lynx adapters loading and sampling on **Wan2.2 i2v A14B** (no VACE in the graph at all), fp16 experts, via an API-submitted workflow. Loader output:

```
Lynx full reference adapter detected
Lynx lite IP adapter detected
Model cross attention type: t2v
Model variant detected: i2v_14B_2.2
```

and at sample time, on both experts:

```
Using Lynx embeddings
Extracted 40 cond ref buffers / 40 uncond ref buffers
```

**Why it works architecturally**, which I didn't see spelled out anywhere: Lynx's ip adapter patches *cross*-attention. Wan2.2 i2v dropped the CLIP image encoder, so `blocks.N.cross_attn` contains only `q,k,v,o,norm_q,norm_k` — no `k_img`. In `nodes_model_loading.py` neither the `k_img` branch nor the `in_channels == 16` branch fires, so `model_type` stays at its initial `"t2v"` — i.e. the same cross-attn shape Lynx was trained against. Tensor geometry lines up too: 2.2 A14B is dim 5120 / 40 blocks, and the adapters are ref `5120x5120` on all 40 blocks, ip `5120x2048` on 20 alternating blocks.

Interesting corollary: Wan2.1 **i2v** *does* carry `k_img` (that's how the loader detects i2v), so for the ip adapter specifically, 2.2 i2v is arguably a *closer* match to Lynx's training setup than 2.1 i2v is.

**One thing specific to the 2.2 dual-expert case** that's separate from the VACE wiring issue resolved above: each expert is its own `WanModel`, and the sampler's check is per-transformer (`transformer.blocks[0].cross_attn.ip_adapter is None`, `nodes_sampler.py:1008`). So the Lynx `extra_model` chain has to reach **both** the high-noise and low-noise model loaders — attach it to only one and you get the #1418 error the moment the sampler reaches the other.

Also, for anyone using an fp8 base: it won't work with Lynx. The adapters are plain `nn.Linear` in `lynx/modules.py`, so they aren't wrapped by the fp8-aware linear; with an fp8 base their weights cast to fp8 while activations stay fp16 and the sampler dies with `self and mat2 must have the same dtype, but got Half and Float8_e4m3fn`. Both `quantization="disabled"` and `"fp8_e4m3fn_scaled"` fail identically, so the checkpoint dtype is the lever rather than the quantization setting. fp16 works.

@Rudra-ai-coder — your "consistency of the faces has increased tremendously" matches what I'm chasing. Still calibrating ip/ref scales on my end; happy to report numbers back if useful.

**Comment by DavidJBarnes** [comment]
Follow-up with measured numbers, plus one gotcha that cost me two runs and is worth flagging for anyone combining Lynx with i2v.

**Aspect stretch — silent, and it hurts identity scores.**

`WanVideoImageToVideoEncode` resizes the start image with:

```python
common_upscale(start_image.movedim(-1,1), W, H, "lanczos", "disabled")   # nodes.py
```

Crop mode is `"disabled"` — a plain resize with no aspect preservation. If your generation
resolution doesn't match the start image's aspect, the frame is stretched silently. A 768x1024
(0.750) start frame generated at 480x832 (0.577) comes out **1.30x stretched vertically**. It looks
subtly wrong, and because ArcFace is geometry-sensitive it also *depresses* any identity metric you
compute. Deriving the generation size from the start image's aspect (snapped to /16) fixed it —
768x1024 -> 624x832 is exact.

**Splitting the two adapters' sources.**

The two adapters want different inputs, and `WanVideoAddLynxEmbeds` takes them separately
(`lynx_ip_embeds` vs `ref_image`), so they can come from two different `LynxInsightFaceCrop`
instances:

- **ip** (ArcFace embedding) encodes *who*, and is largely pose/lighting-invariant — feed it a
  canonical high-res portrait.
- **ref** (dense VAE features) encodes *appearance* — feed it a crop of the **start frame**, so it
  matches the shot instead of fighting it.

Measured on Wan2.2 i2v A14B, fp16 experts, de-distilled (20 steps split 10/10, cfg 3.5, shift 5.0),
29 frames, cosine vs the canonical portrait over 5 interior frames:

| config | mean | drift across clip |
|---|---|---|
| both adapters from canonical photo | 0.560 | -0.091 |
| split sources, ip_scale 0.7 | 0.614 | -0.065 |
| split sources, ip_scale 1.0 | **0.625** | **-0.026** |

The mean difference between ip 0.7 and 1.0 is small (n=1, not significant), but the *drift* trend is
consistent: higher ip_scale keeps the trace flat where 0.7 slides monotonically. That matches the
mechanism — the ip adapter re-asserts identity every step, so stronger re-assertion resists
accumulated drift. @Rudra-ai-coder this may be what you were seeing qualitatively.

**Caveat, stated plainly:** these are all Lynx-vs-Lynx comparisons. I have not yet run a **no-Lynx
control**, so none of the above proves Lynx beats plain i2v — on a short, near-frontal clip,
start-frame conditioning alone might do as well. Nor have I tested interaction with motion LoRAs,
which load on the low-noise expert while Lynx's ref adapter patches self-attention on every block of
both experts. Treat these as calibration data, not as evidence of effect size.
